documentation/docs/protocol_specification/components/congestion_control.md

```diff
@@ -1,87 +1,186 @@
+this rate is enforceable, and that increasing delays or dropped messages are only experienced by misbehaving node.
+However, it is unrealistic that all nodes will always have messages to issue, and we would like nodes to better
+utilise network resources, without causing excessive congestion and violating any requirement.
+
+We propose a rate setting algorithm inspired by TCP — each node employs [additive increase, multiplicative decrease]
+(https://https://epubs.siam.org/doi/book/10.1137/1.9781611974225) (AIMD) rules to update their issuance rate in
+response to congestion events. In the case of distributed ledgers, all message traffic passes through all nodes,
+contrary to the case of traffic typically found in packet switched networks and other traditional network
+architectures. Under these conditions, local congestion at a node is all that is required to indicate congestion
+elsewhere in the network. This observation is crucial, as it presents an opportunity for a congestion control
+algorithm based entirely on local traffic.
+
+Our rate setting algorithm outlines the AIMD rules employed by each node to set their issuance rate. Rate updates
+for a node `node` take place each time a new message is scheduled if the `node` has a non-empty set of its own
+messages not yet scheduled. Node `node` sets its own local additive-increase variable `localIncrease(node)` based on
+its access Mana and on a global increase rate parameter `RATE_SETTING_INCREASE`. An appropriate choice of
+`RATE_SETTING_INCREASE` ensures a conservative global increase rate which does not cause problems even when many
+nodes increase their rate simultaneously. Nodes wait `RATE_SETTING_PAUSE` seconds after a global multiplicative
+decrease parameter `RATE_SETTING_DECREASE`, during which there are no further updates made, to allow the reduced
+rate to take effect and prevent multiple successive decreases. At each update, `node` checks how many of its own
+messages are in its outbox queue, and responds with a multiplicative decrease if this number is above a threshold,
+`backoff(node)`, which is proportional to `node`'s access Mana. If the number of `node`'s messages in the outbox is
+below the threshold, `node`'s issuance rate is incremented by its local increase variable `localIncrease(node)`.
+
+### Tip pool and Time Since Confirmation check
+
+When a message is scheduled, it's added to the tip pool, however there are additional actions and checks that need
+to be performed before a message is added:
+
+* a confirmed message shall not be added to the tip pool (it shall be skipped by the scheduler)
+* a message that has confirmed or scheduled approvers shall not be added to the tip pool
+* strong parents of a message are removed from the tip pool, when the message is added
+* after certain amount of time, unused tips are removed from the tip pool
+
+When selecting tips from the tip pool an additional check shall be performed to make sure that the timestamp and a
```

Comment:
See my comment above: I think this should be explained in the TSA spec with more details.

Reply:
Should I wait for this or can I go ahead and review?

Reply:
should I wait for this or can I go ahead and review this section as well?

Reply:
I will move it and then I'll re-request your review.